### DISCUSSION TITLE
Live Editor

### DISCUSSION THREAD

**Discussion by Kenshin** [Kenshin/simpread#2852] [插件说明]
描述
---

> 为当前页面增加 Live Editor（实时编辑）

<img width="808" alt="pika-1745570648090-1x@cover" src="https://github.com/user-attachments/assets/29015acb-bbc2-4033-a3de-b3fb308cfb54" />

地址
---

[https://simpread.ksria.cn/plugins/details/y8Mai5IBwN](https://simpread.ksria.cn/plugins/details/y8Mai5IBwN)

功能
---

1. [Live Editor  2.0](https://github.com/Kenshin/simpread/discussions/2852#discussioncomment-1976860)

   <img width="507" alt="image" src="https://github.com/user-attachments/assets/c766fdfc-81f8-4939-99c8-fb422278f5e7" />

2. [简易模式](https://github.com/Kenshin/simpread/discussions/2852#discussioncomment-2489840)

3. [清除元素](https://github.com/Kenshin/simpread/discussions/2852#discussioncomment-7020902)

4. [居中显示](https://github.com/Kenshin/simpread/discussions/2852#discussioncomment-7020862)

5. [复制到原文](https://github.com/Kenshin/simpread/discussions/2852#discussioncomment-1976817)

6. [添加/删除描述](https://github.com/Kenshin/simpread/discussions/2852#discussioncomment-12944459)

选项
---

随便进入一个页面 e.g. https://sspai.com/post/71576 并进入 **阅读模式 → 右下角 → 动作 → 插件触发器**

按照下图所示即可查看此插件的选项页。（注意：如果你的界面没有插件触发器，请根据此 [解决方案](https://github.com/Kenshin/simpread/discussions/2342) 修复。

<img width="1294" alt="image" src="https://github.com/user-attachments/assets/af7f3215-4d26-4862-996e-a0a958d09ba9" />

设置项
---

- 进入阅读模式后自动加载工具栏

    > 选否后，可以在**阅读模式 →  控制栏 → 动作 → 插件触发器**手动进入

    <img width="526" alt="image" src="https://github.com/user-attachments/assets/e5ecb6ff-3a85-478b-9b1e-8bdd0bab4f18" />

- 当加载工具栏后自动进入 Live Editor 模式

    > 不建议开启

    <img width="528" alt="image" src="https://github.com/user-attachments/assets/c8d4a70a-3872-49be-8bba-529f65b88199" />

- 当使用【复制到原文】后自动退出阅读模式

     > 通过此功能可以弥补其它剪藏工具的不足，如：[导入到 Zotero](https://github.com/Kenshin/simpread/discussions/5927)
    
     <img width="524" alt="image" src="https://github.com/user-attachments/assets/e87c2402-4fb4-4e3d-95b1-a554d073f8c6" />

- 自动保存快照到本地

    > 当退出【编辑模式】【清除元素】【居中显示】后自动保存快照到本地，仅限当前页面为稍后读时可用。如果你配置了 [知识库](https://www.yuque.com/kenshin/simpread/wkswh7) 功能后建议开启此选项。

    <img width="524" alt="image" src="https://github.com/user-attachments/assets/08f48d94-5bb4-43e3-b97f-0e9014eabe0f" />

快捷键
---

<kbd>l e</kbd> + <kbd>1 ~ 6</kbd> 的标号，编号对应了下面的工具条的顺序

<img width="732" alt="image" src="https://github.com/user-attachments/assets/1aa7160f-7278-4d24-9807-c5339e290401" />

比如：<kbd>l e 1</kbd> 对应了 **开启/关闭 Live Editor**

更多用法
---

- [基于简悦 + Live Editor 的的信息收集与多端处理工作流](https://zhuanlan.zhihu.com/p/453792136)

- [如何用简悦插件 Live Editor 快速采集和标注网页图文内容？](https://www.zhihu.com/zvideo/1424673951406166016)

- [将需要的任意内容生成阅读模式并使用 Live Editor 进一步整理](https://github.com/Kenshin/simpread/discussions/3233)

- [导入到 Zotero（利用简悦增强 Zotero 的剪藏能力）](https://github.com/Kenshin/simpread/discussions/5927)

关于 Live Editor 更多用法可以 [看这里](https://github.com/Kenshin/simpread/discussions?discussions_q=label%3A%22live+editor%22)。

安装与升级
---

- 没有【使用同步助手 · 自动同步】的话，可通过 [此链接](https://simpread.ksria.cn/plugins/details/y8Mai5IBwN) 安装。

- 已经【使用同步助手 · 自动同步】的话，请使用 [插件管理器](https://www.yuque.com/kenshin/simpread/fcw60u#ymS1C) 安装。

更新日志
---

### 2.0.0

- `Add` 全新的 [Live Editor 2.0](https://github.com/Kenshin/simpread/discussions/2852#discussioncomment-1976860)

- `Add` [添加/删除描述段落](https://github.com/Kenshin/simpread/discussions/2852#discussioncomment-12944459)

- `Optimize` 选项卡 UI 2.0

### 1.2.0

- `Add` 居中显示，通过框选的方式将图片或段落居中显示，包括：`图片` 和 `任意段落`

- `Add` 开启 `简易模式` 或 `Live Editor` 后取消连续标注功能

- `Add`  自动保存快照到本地

### 1.1.4

- `Fix` 当使用清除元素功能后图片标注失效的错误。 

### 1.1.3

- `Add` 简易编辑模式，不会破坏正文结构，但也无法提供富文本编辑功能。

### 1.1.2

- `Add` 快捷键支持

- `Add` 复制到原文

- `Add` 存在标注时，仅使用  **简单编辑模式**

- `Optimize` （不存在标注的）稍后读页面也可使用 Live Editor 功能

- `Optimize` `blockquote` 样式

- `Fix` 退出 Live Editor 后，加入稍后读后无法使用标注功能

- `Fix` 在 Live Editor 下无法使用 [题图](https://simpread.ksria.cn/plugins/details/PcmPCT9rgM)

### 1.1.1

- `Fix` 进入 Live Editor 后样式错误

- `Fix` 标注模块能可以使用的错误

### 1.1.0

- `Add`  `[[<选取的文字>]]`

- `Add` 复制所选内容到剪切板

- `Fix` 当存在标注，进入时提示，并且隐藏标注侧栏；退出时，恢复标注侧栏并提示需要刷新页面。

- `Fix` 当标注模式下，进入 Live Editor 需要禁用标注模式

- `Add` 选取文字生成 Markdown，类似 roam-highlighter 可以根据上下文生成的方案。

    - 划取后的内容用颜色标记出来

    - 根据上下文层级整理标记

    - 点击已标记的段落后删除当前标记

    - 在页面右下角显示，包括：复制到剪切板，最小化，清除全部标记

    - 设置

        - 是否自动去除换行？

        - 当标记的内容为包含 标题 / 列表 时，是否去除？

        - 标记之间是否空一行？

        - 标记后的前缀

### 1.0.0

- 下方工具条

    - 开启/关闭 Quill

    - 删除段落模式

    - 设置

    - 退出

- 设置

    - 进入阅读模式后是否自动加载工具栏？

    - 当自动加载工具栏后，是否自动进入 Live Editor？

**Comment by Kenshin** [comment]
# 全新 2.0 版已废除此功能

<details><summary><h2>1.x 版功能说明<h2></summary>

# 标记功能

> 为当前页面提供一种：**轻量级，方便导入到各种双链笔记的方案。**

![image](https://user-images.githubusercontent.com/81074/133882588-4434ee85-bd06-4a3e-9f92-e8c711449b9c.png)

## 与简悦的标注功能区别

1. 不具有简悦标注功能的相关一切功能，如：[导出到本地](http://ksria.com/simpread/docs/#/保存到本地)，[导出到简悦支持的生产力工具](http://ksria.com/simpread/docs/#/导出到生产力工具)，[简悦的双向链接支持](http://ksria.com/simpread/docs/#/稍后读-多种布局?id=双向链接) 等功能。

2. 仅提供一种轻量级操作，即：**标记 → 复制 Markdown 到剪切板 → 粘贴到你需要的各种双链笔记**。

3. 此功能也不支持 [Markdown 定制模板](http://ksria.com/simpread/docs/#/定制化导出?id=markdown)。

## 特色

1. 可标记图片

2. 可根据上下文的顺序进行标记，且具有从属关系。

3. 可随时在正文中修改内容，直至满意后再进行标记。

## 功能

### 在页面选中的内容增加 `[[]]` 操作。

![2021-09-18_15-34-03 (1)](https://user-images.githubusercontent.com/81074/133880628-04ca2e54-840e-4b08-9b34-957fc4d00534.gif)

### 标记后将会生成适合导入到双链笔记的 Markdown 语法规则

![2021-09-18_15-39-43 (1)](https://user-images.githubusercontent.com/81074/133880816-b2654046-5e2e-4d12-93c2-78df534771c3.gif)

### 对已经标记的重新标记即为删除

![2021-09-18_15-48-18 (1)](https://user-images.githubusercontent.com/81074/133881025-fe9bf638-ab0e-4c79-b44b-e8e2e55156e0.gif)

### 关于标记功能的一些选项

![image](https://user-images.githubusercontent.com/81074/133881080-a0403274-de6f-4b6d-87c1-4b1c9c2365ef.png)

1. 自动去除换行

2. 自动去除包含 标题 / 列表的标记内容

3. 每个标记之间增加一个空行

4. 标记的前缀，如 `-` `>`

![image](https://user-images.githubusercontent.com/81074/133881193-7bbe5458-e3c4-4d93-aa7f-9d06346449fd.png)

![image](https://user-images.githubusercontent.com/81074/133881243-d9402048-e9fd-4d59-9ba0-7b91fe7d3421.png)

### 注意

> 标记功能与简悦的标注功能有冲突，在存在标注时，进入 Live Editor 的编辑状态后，如果存在标注的话，有无法显示，且有提示。

![image](https://user-images.githubusercontent.com/81074/134755684-da7e0a4b-9544-4d3c-9683-ac5a1d04090d.png)

> 此功能为版本 1.1.1 提供，如低于此版本，请升级。

![image](https://user-images.githubusercontent.com/81074/134755709-aa8c2c23-80d4-4711-bcb2-b244d9abb6ce.png)


</details>

**Comment by angel126** [comment]
为什么标记的颜色无法正常显示？
![image](https://user-images.githubusercontent.com/8450915/134753279-0c2a9ca9-2a0f-4447-b659-e4a248fed8ed.png)

**Reply by Kenshin** [reply]
嗨，请看 https://github.com/Kenshin/simpread/discussions/2852#discussioncomment-1350262 下方的**注意**段落。

简单的说，在此模式下无法显示标注功能。

**Comment by hkdiy** [comment]
无法使用插件阅读模
![Snipaste_2021-09-26_15-31-45](https://user-images.githubusercontent.com/13012269/134798419-e4dc09ba-248c-4061-b6dc-700be3b4bd6c.jpg)
![Snipaste_2021-09-26_15-31-59](https://user-images.githubusercontent.com/13012269/134798430-7e2e7e7e-5d26-46f9-85db-8625b65e13f7.jpg)

并没有live editer选项, 也无法在阅读模式下找到工具栏.
另外安装其他插件也无法
使用.

**Reply by Kenshin** [reply]
安装插件或带有插件触发器的插件失效的 [解决方案](https://github.com/Kenshin/simpread/discussions/2342)。

**Comment by c100d100** [comment]
用Live Editor编辑了文章内容后，保存到稍后读里的文章仍是未编辑过的。

下图是在Live Editor中编辑的内容，我先后尝试了
* 编辑内容后，关闭Live Editor
* 编辑内容后，退出Live Editor
* 右键菜单保存到稍后读
* 动作 - 保存到稍后读
* 快捷键保存到稍后读

![1](https://user-images.githubusercontent.com/93242004/139004551-f1485334-0258-4efc-b05b-89484be14590.jpg)

但结果都是像下图这样，保存到稍后读的文章仍是一点都未经修改的样子。

<img width="930" alt="2" src="https://user-images.githubusercontent.com/93242004/139004523-d2ca2a2d-977a-4ea2-8397-7bef701465a4.png">

但神奇的是，同步保存到notion里的文章却是正常的：

<img width="710" alt="3" src="https://user-images.githubusercontent.com/93242004/139007181-7b3975d0-eece-426a-9095-284f5af3fe17.png">

**Reply by alexinea** [reply]
使用 Live Editor 编辑后：
1、退出 LE
2、加入稍后读
3、在网页版的页面里任意标注
4、此时稍后读显示的是编辑后的效果

**Reply by Kenshin** [reply]
@alexinea 

有什么问题吗？

**Reply by Kenshin** [reply]
@alexinea 

emmm 好像是回复 @c100d100 的提问？

> 如果是的话，请忽略我的上一条内容，并谢谢你的回复。

💪

PS：漏掉了 @c100d100 的提问...

**Comment by Kenshin** [comment]
占位

**Comment by Kenshin** [comment]
# 复制到原文

简单的说，通过此功能，你可以将阅读模式的内容（正文）复制到原页面。

## 使用场合

当你使用其它 Web Cliper 时，可利用此功能辅助加强其它 Web Cliper 的正文获取能力，如：[导入到 Zotero](https://github.com/Kenshin/simpread/discussions/5927)

## 操作视频

https://user-images.githubusercontent.com/81074/149648587-9e04381d-5beb-4a2c-8e79-708578c32f99.mp4

## 选项

一个相关的选项：当复制到原文后，可自动退出阅读模式。

![image](https://github.com/Kenshin/simpread/assets/81074/c04c49b9-25f2-4f22-ae81-adaa1d4186c0)

**Comment by Kenshin** [comment]
# Live Editor 2.0

![liveeditor2@origin_background](https://github.com/user-attachments/assets/bda28c9c-881d-4660-8a01-e48152d0d055)

## 写在前面

因为简悦支持标注，所以在使用上会跟富文本编辑冲突，所以在 1.x 版去掉了此功能。

2.0 版我重写了富文本编辑功能，因为使用了自有方案，而非第三方 Library，所以可以完美的兼容简悦的标注系统。

## 2.0 版

功能包括：粗体 / 斜体 / 下划线 / 删除线 / 引用 / 代码块 / H1 ~ H3 / 清除样式

## 如何使用

安装 Live Editor 后在页面下方会有个工具条，点击 Live Ediot 2.0 即可开始进行富文本编辑

<img width="741" alt="2025-04-25_15-06-12-2" src="https://github.com/user-attachments/assets/1e0cfbbf-b13b-4271-82bf-9dd4ccf28760" />

## 视频演示

https://github.com/user-attachments/assets/9dbac934-d129-4f61-a5f1-1a76c7d6470f

**Reply by liukmg** [reply]
您好，想请问一下编辑模式下加粗样式显示不正常是什么原因呢？进入阅读模式之后点击打开live editor的按钮，选中文字在工具栏点击加粗，但是字体会变小，没有显示加粗的效果。退出live editor之后阅读模式也还是变小的字体，加入稍后读之后不管从网页还是logseq侧边栏打开都还是显示不正常。
比如下面这个链接，这个里面除了加粗不正常，标题字体显示也不太正常，而且打开live editor之后最下面的一张对比的大表也消失了。
https://www.ppmfunds.com/imas-and-smas-explained/

一些基础信息：简悦高级账户，使用的是Chrome浏览器，版本 114.0.5735.199（正式版本） （64 位）。已经按照简悦手册中的一站式配置指南做好了所有配置，都可以正常运行，只是live editor显示不太正常。

期待您的回复，非常感谢！如果有任何需要补充的信息请随时联系我，再次感谢！ @Kenshin

**Reply by Kenshin** [reply]
> 选中文字在工具栏点击加粗，但是字体会变小

这个功能并没有那么 100% 可以使用，因为不同的网页有不同的「规则」。

我用你给的 URL 随机测试了几个地方，是没问题的。

<img width="381" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/5f5154bb-9f12-4866-a6cd-0a0435d54424">

所以：

1. 不建议纠结这个问题。
2. 截图指出具体哪里设置后没有加粗？

**Reply by liukmg** [reply]
感谢您的时间！确实是小问题，能解决最好，解决不了就暂时不用加粗这个样式也可以。

这是打开阅读模式之后什么都不点的样子：
![86708c98fe4f3e5b134c78294397663](https://github.com/Kenshin/simpread/assets/44069048/130ea5fc-3c7f-4645-ac83-43b7be780e5e)

这个是打开了live editor之后的样子：
![e2abe33b60a7072a3f698e3ef70409d](https://github.com/Kenshin/simpread/assets/44069048/698328a7-ff3f-45f4-9d25-0e57efd9be92)

这个是加粗了红色这句话之后的样子：
![cc69b27680337819dfc885ce4eebcec](https://github.com/Kenshin/simpread/assets/44069048/056ed204-d813-4f00-b6a6-6b0322def375)
![cc9b8e374861afe5aa9fb0d6335c49f](https://github.com/Kenshin/simpread/assets/44069048/c20b0d40-386e-411d-acfb-09862f3077ae)

**Reply by Kenshin** [reply]
emmm 看截图似乎跟划词翻译「冲突」了，你试试先禁用划词翻译。

<img width="471" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/8aefd249-3367-40ae-aa00-f67531c3dc10">

然后重启浏览器后再试试。

**Reply by liukmg** [reply]
关掉了试也还是不行诶，三张截图和上面一样的，没有什么变化。昨晚上我没有安装划词翻译的插件之前已经是这样了，不知道是什么疑难杂症。也不急，您后续开发的时候如果偶尔猜到了可能是什么原因可以随时联系我来试试看，感谢！

**Reply by Kenshin** [reply]
okay 主要是仍没有遇到你提到的问题，但安装划词翻译确实会影响 Live Editor 😂

**Reply by liukmg** [reply]
好滴，我又试了一下，有一些网站live editor的编辑功能里面的样式基本都是乱套的，不只是加粗，可能是和我其他的什么插件冲突了吧，非常个性化的问题了，我自己也摸索一下，感谢您的时间！不过好在编辑文字、删除元素都没问题，以及解决我主要的需求了~

**Reply by Kenshin** [reply]
Live Editor 在使用时确实会有将原页面「弄乱」的可能行，这也是我之前前面提到的「并不太好使」的原因。

一般结构良好的页面大多没问题。

**Comment by Smallart** [comment]
@Kenshin  您好
**环境**

操作系统：Win10
浏览器：Edge  版本 97.0.1072.69
简悦版本 2.2.0
live editor插件版本 1.1.2

**问题**

进入编辑模式后，对文章 ctrl+v 黏贴一些内容，页面会从黏贴的地方闪回到文章的开头。我尝试在 InPrivate 模式下测试，依然有这问题。这个是快捷键冲突吗？

**Reply by Kenshin** [reply]
嗨，这是个 Bug，但具体如何产生还需要我研究下。

## 临时解决方案

当 `ctrl + v` 之后使用箭头（如使用 <kbd>←</kbd> 就可以重新回到粘贴内容的位置上。

**Comment by Kenshin** [comment]
# 简易模式

1.1.3 版新增加的功能，特点是：不会破坏正文结构，但也无法提供富文本编辑功能。

## 位置

<img width="700" alt="image" src="https://github.com/user-attachments/assets/3d4031c0-990f-46ac-bc74-e679d7c1ae52" />

## 操作视频

https://user-images.githubusercontent.com/81074/161367410-932e5f99-8063-4db0-b946-15400ff12ae8.mp4

## 注意

1. 因为简悦的标注功能根据上下文进行定位的，如果对已存在标注的「前面」进行修改的话，会导致标注定位失败。

2. 当存在标注的页面使用 Live Editor 后会自动进入此模式。

    https://user-images.githubusercontent.com/81074/161673098-7367f693-173c-4ef1-bbde-ff472c2959cd.mp4

**Comment by Kenshin** [comment]
# 居中显示

1.2.0 版新增加的功能，包括：图片和任意段落，选中后可居中显示。

## 位置

![image](https://github.com/Kenshin/simpread/assets/81074/4642f202-916a-405f-aab9-7970d8bb7190)

## 操作视频

https://github.com/Kenshin/simpread/assets/81074/e3384a52-ad09-4973-982c-4ec8338bbf5f

**Comment by Kenshin** [comment]
# 清除元素

可通过框选的方式删除任意内容。

## 位置

<img width="700" alt="image" src="https://github.com/user-attachments/assets/2e156079-226e-4ec1-9e95-90c640efbc66" />

## 视频操作

![2022-01-16_14-11-32 (1)](https://user-images.githubusercontent.com/81074/149649453-b47f4909-0674-4582-8556-5ed67c41c9b4.gif)

**Comment by Kenshin** [comment]
# 添加删除描述

<img width="705" alt="image" src="https://github.com/user-attachments/assets/0d772ddc-8352-4d38-a6df-0b08bc371269" />

## 功能描述

这是 2.0 版增加的功能，方便对当前页面添加/删除描述段落。

## 视频操作

https://github.com/user-attachments/assets/394d7d96-ebc0-476e-90a2-02be30bc29c5
